### PR TITLE
[FlaxBart] make sure no grads are computed an bias

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

Makes sure that no gradients are computed on FlaxBart bias. Note that in PyTorch we don't compute a gradient because the bias logits are saved as a PyTorch buffer: https://discuss.pytorch.org/t/what-is-the-difference-between-register-buffer-and-register-parameter-of-nn-module/32723

Also see results of: https://discuss.huggingface.co/t/gradients-verification-between-jax-flax-models-and-pytorch/15970

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
@patil-suraj @sanchit-gandhi - this surely makes a difference when training - I'm not sure how much of a difference though. Will be interesting to try out though.

**Comment by sanchit-gandhi** [comment]
The `stop_gradient` fix looks good - the results you cited verify nicely that the gradients are only frozen for the `final_logits_bias` parameters, and not for any other additional parameters upstream in the computation graph.

One small comment in regards to the testing script `check_gradients_pt_flax.py` that you used to generate the gradient comparison results. Currently, we are comparing the PyTorch gradients relative to the Flax ones (https://huggingface.co/patrickvonplaten/codesnippets/blob/main/check_gradients_pt_flax.py#L86):
```python
diff_rel = np.abs(ak_norm - bk_norm) / np.abs(ak_norm)
```
Since we are taking the PyTorch gradients as our ground truth values, we should compare the Flax gradients relative to the PyTorch ones:
```python
diff_rel = np.abs(ak_norm - bk_norm) / np.abs(bk_norm)
```

**Review by sanchit-gandhi — APPROVED** [review]

**Review by patil-suraj — APPROVED** [review]
That's great catch ! Thanks a lot for diving into this
